### PR TITLE
CRDS-1941: dedupe sds plus logic

### PR DESCRIPTION
**Pull request by james-baker-digital-justice**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonApiDataMapper.kt [hunk 1]
```diff
@@ -4,31 +4,32 @@ import com.fasterxml.jackson.core.type.TypeReference
...
-        sentencesAndOffences.map { it.toLatest() }
+        sentencesAndOffences.flatMap { it.toLatest() }
       }
       1 -> {
         val reader = objectMapper.readerFor(object : TypeReference<List<PrisonApiSentenceAndOffences>>() {})
-        reader.readValue<List<PrisonApiSentenceAndOffences>>(calculationRequest.sentenceAndOffences).map { SentenceAndOffencesWithReleaseArrangements(it, false) }
+        reader.readValue<List<PrisonApiSentenceAndOffences>>(calculationRequest.sentenceAndOffences).flatMap { prisonApiSentenceAndOffences ->
+          prisonApiSentenceAndOffences.offences.map { offence -> SentenceAndOffenceWithReleaseArrangements(prisonApiSentenceAndOffences, offence, false) }
+        }
       }
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonApiDataMapper.kt:30]: Not a fan of the toLatest approach? ;)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenceSdsPlusLookupService.kt [hunk 2]
```diff
@@ -93,10 +97,10 @@ class OffenceSdsPlusLookupService(
         offence.indicators = offence.indicators.plus(listOf(OffenderOffence.SCHEDULE_15_LIFE_INDICATOR))
         sdsPlusIdentified = true
       } else if (sentenceIsAfterPcsc && sevenYearsOrMore && moResponseForOffence?.pcscMarkers?.inListD == true) {
-        offence.indicators = offence.indicators.plus(listOf(OffenderOffence.PCSC_SDS))
+        offence.indicators = offence.indicators.plus(listOf(OffenderOffence.PCSC_SDS_PLUS))
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/OffenceSdsPlusLookupService.kt:100]: These are probably not worth setting now are they?

**Comment by srowlands-moj** [comment]
LGTM - shall await phase 2 with bated breath

**Review by srowlands-moj — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonApiDataMapper.kt (hunk 1 above)
Comment (james-baker-digital-justice) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/PrisonApiDataMapper.kt]: Fixed this to match

**Review by srowlands-moj — APPROVED** [review]
LGTM

**Pull request merged**